src/app/hooks/keycloak.ts

```diff
@@ -13,18 +13,19 @@ const init = async (config: Config, getInsightsAccessToken: () => Promise<string
   return await initKeycloak(keycloakConfig, getInsightsAccessToken);
 }
 
-export const useAuth = (): Auth => {
+export const useAuth = (): Auth | undefined => {
```

Comment:
How will you otherwise tell the components that useAuth that Auth is not yet loaded because config is not yet loaded too?
Note: my understanding is that is a bad practice to block the react render

Reply:
The methods on auth return an undefined if the authentication is not ready yet now - see https://github.com/redhat-developer/app-services-ui-shared/blob/main/src/contexts/auth/Auth.ts

This allows us to construct the component tree, and not block rendering on things not being ready, and have data loading wait for what it needs.

Reply:
Make sense, will convert it then